chainerrl/agent.py

```diff
@@ -173,3 +173,68 @@ def process_idx(self):
     def shared_attributes(self):
         """Tuple of names of shared attributes."""
         pass
+
+
+class BatchAgent(with_metaclass(ABCMeta, Agent)):
+    """Abstract agent class that can interact with a batch of envs."""
+
+    @abstractmethod
+    def batch_act(self, batch_obs):
+        """Select a batch of actions for evaluation.
+
+        Args:
+            batch_obs (Sequence of ~object): Observations.
+
+        Returns:
+            Sequence of ~object: Actions.
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def batch_act_and_train(self, batch_obs):
+        """Select a batch of actions for training.
+
+        Args:
+            batch_obs (Sequence of ~object): Observations.
+
+        Returns:
+            Sequence of ~object: Actions.
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def batch_observe(self, batch_obs, batch_reward, batch_done, batch_reset):
```

Comment:
Being not allowed to learn here, agents don't have to take `batch_reward` and `batch_done` in the `batch_observe` method.  I'm not sure if `batch_obs` can be removed.

Reply:
As we discussed offline, I keep the same arguments to make the API consistent between training and evaluation.